Core/tests/ObjectsContainersList.cpp

```diff
@@ -0,0 +1,488 @@
+                           "MyOtherBehavior");
+    gd::Object &object2 = layout.GetObjects().InsertNewObject(
+        project, "MyExtension::Sprite", "MyObject2", 0);
+    object2.AddNewBehavior(project, "MyExtension::MyBehavior", "MyBehavior");
+
+    auto &group = layout.GetObjects().GetObjectGroups().InsertNew("MyGroup", 0);
+    group.AddObject(object1.GetName());
+    group.AddObject(object2.GetName());
+
+    auto objectsContainersList = gd::ObjectsContainersList::
+        MakeNewObjectsContainersListForProjectAndLayout(project, layout);
+
+    REQUIRE(objectsContainersList.HasBehaviorInObjectOrGroup("MyGroup", "MyBehavior"));
+    REQUIRE(!objectsContainersList.HasBehaviorInObjectOrGroup("MyGroup", "MyOtherBehavior"));
+  }
+}
+
+TEST_CASE("ObjectContainersList (GetBehaviorsOfObject)", "[common]") {
+
+  SECTION("Find the behaviors in an object") {
+    gd::Platform platform;
+    gd::Project project;
+    SetupProjectWithDummyPlatform(project, platform);
+
+    gd::Layout &layout = project.InsertNewLayout("Scene", 0);
+    gd::Object &object = layout.GetObjects().InsertNewObject(
+        project, "MyExtension::Sprite", "MyObject", 0);
+    object.AddNewBehavior(project, "MyExtension::MyBehavior", "MyBehavior");
+
+    auto objectsContainersList = gd::ObjectsContainersList::
+        MakeNewObjectsContainersListForProjectAndLayout(project, layout);
+
+    const auto behaviors =
+        objectsContainersList.GetBehaviorsOfObject("MyObject", true);
+    REQUIRE(behaviors.size() == 1);
+    REQUIRE(behaviors[0] == "MyBehavior");
+  }
+
+  SECTION("Find the behaviors in a group") {
+    gd::Platform platform;
+    gd::Project project;
+    SetupProjectWithDummyPlatform(project, platform);
+
+    gd::Layout &layout = project.InsertNewLayout("Scene", 0);
+    gd::Object &object1 = layout.GetObjects().InsertNewObject(
+        project, "MyExtension::Sprite", "MyObject1", 0);
+    object1.AddNewBehavior(project, "MyExtension::MyBehavior", "MyBehavior");
+    // This behavior is only in one of the 2 objects.
+    object1.AddNewBehavior(project, "MyExtension::MyOtherBehavior",
+                           "MyOtherBehavior");
+    gd::Object &object2 = layout.GetObjects().InsertNewObject(
+        project, "MyExtension::Sprite", "MyObject2", 0);
+    object2.AddNewBehavior(project, "MyExtension::MyBehavior", "MyBehavior");
+
+    auto &group = layout.GetObjects().GetObjectGroups().InsertNew("MyGroup", 0);
+    group.AddObject(object1.GetName());
+    group.AddObject(object2.GetName());
+
+    auto objectsContainersList = gd::ObjectsContainersList::
+        MakeNewObjectsContainersListForProjectAndLayout(project, layout);
+
+    const auto behaviors =
+        objectsContainersList.GetBehaviorsOfObject("MyGroup", true);
+    REQUIRE(behaviors.size() == 1);
+    REQUIRE(behaviors[0] == "MyBehavior");
+  }
+}
+
+namespace {
+
+gd::SpriteObject BuildSpriteWithAnimations(gd::String animationName1 = "!",
```

Comment:
Curious: any particular reason not to use an empty string as the special value?

Reply:
I thought that an animation with an empty name is something that is often used and we'll want to test it at some point.